bundler/lib/bundler/lockfile_generator.rb

```diff
@@ -65,6 +66,22 @@ def add_dependencies
       end
     end
 
+    def add_checksums
+      out << "\nCHECKSUMS\n"
+      definition.resolve.sort_by(&:full_name).each do |spec|
+        checksum = spec.to_checksum if spec.respond_to?(:to_checksum)
+        if spec.is_a?(LazySpecification)
+          spec.materialize_for_checksum do |materialized_spec|
+            checksum ||= materialized_spec.to_checksum if materialized_spec&.respond_to?(:to_checksum)
+          end
+        end
+
+        checksum ||= definition.locked_checksums.find {|c| c.match_spec?(spec) }
```

Comment:
this is going to be quadratic -- if we make a hash keyed by full name, we can avoid looping through the full array of locked checksums